docs/source/development/start/setup.rst

```diff
@@ -0,0 +1,165 @@
+
+.. _contribution_install:
+
+Installation
+------------
+
+Cloning Repo
+************
+
+First `fork <https://docs.github.com/en/get-started/quickstart/fork-a-repo>`_ the ``lineapy`` repo.
+Then, `clone <https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository>`_
+the forked repo to your development environment:
+
+.. code:: bash
+
+    cd <PATH-TO-DESIRED-LOCATION>
+    git clone <URL-TO-FORKED-REPO>
+
+To keep the forked repo in sync with the original one, set an "upstream":
+
+.. code:: bash
+
+    git remote add upstream https://github.com/LineaLabs/lineapy.git
+
+Setting up Virtual Environment
+******************************
+
+When done cloning, move into the repo and initiate a virtual environment:
+
+.. code:: bash
+
+    cd <FORKED-REPO-NAME>
+    python -m venv env
+
+where ``<FORKED-REPO-NAME>`` is ``lineapy`` unless modified during/after forking.
+
+This creates ``env/`` subfolder inside the repo, and you can activate the associated virtual environment like so:
+
+.. code:: bash
+
+    source env/bin/activate
+
+To deactivate the virtual environment, you can type:
+
+.. code:: bash
+
+    deactivate
+
+Installing Dependencies
+***********************
+
+Once the virtual environment is activated, install packages necessary for running and developing ``lineapy``:
+
+.. code:: bash
+
+    pip install --upgrade pip
+    pip install --upgrade setuptools
+    pip install -r requirements.txt
+    pip install -e '.[dev]'
+
+Note that this may take a while to complete.
+
+Running LineaPy
+---------------
+
+During development, you may want to check how your changes affect the behavior of the package.
+LineaPy supports several interfaces to run on, and the relevant instructions are documented :ref:`here <interfaces>`.
+Please select the most convenient option for you.
+
+.. note::
+
+    When running LineaPy, make sure that the virtual environment has been activated.
+
+Development
+-----------
+
+Making Changes
+**************
+
+With development dependencies installed, you are now ready to contribute to the source code!
+First, make a separate branch for your development work. Please use an informative name so that
+others can get good sense of what your changes are about.
+
+.. code:: bash
+
+    git checkout -b <NEW-BRANCH-NAME>
+
+After making changes you desire, save them to your development branch:
+
+.. code:: bash
+
+    git add <PATH-TO-CHANGED-FILE>
+    git commit -m "<COMMIT-MESSAGE>"
+
+.. note::
+
+    To learn more about saving changes in Git, check this `tutorial <https://www.atlassian.com/git/tutorials/saving-changes>`_.
+
+Note that these changes have been saved only locally at this point, and you need to "push" them to your forked repo on GitHub:
+
+.. code:: bash
+
+    git push
+
+If the new (development) branch has not been pushed before, you will need to create its counterpart on GitHub with:
+
+.. code:: bash
+
+    git push --set-upstream origin <NEW-BRANCH-NAME>
```

Comment:
actually origin? Shouldn't we be telling contributors to fork?

Reply:
oh i see, we want origin to be their fork and our version to be a named upstream ... 

Its safe to assume people who know how to set remote repos will know to change this command right?

Reply:
Yeah, I think having the "original" repo as `upstream` and the forked repo as `origin` is quite standard a practice as per [GitHub instructions](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-branch-with-the-github-cli). Plus, we have other related instructions use consistent references. If this confuses folks, however, we can revisit this.